iceoryx2/tests/service_tests.rs

```diff
@@ -53,6 +63,230 @@ mod service {
+        let service_name = generate_name();
+
+        std::thread::scope(|s| {
+            let mut threads = vec![];
+            for _ in 0..NUMBER_OF_THREADS {
+                threads.push(s.spawn(|| {
+                    for _ in 0..NUMBER_OF_ITERATIONS {
+                        barrier_enter.wait();
+
+                        let sut = Factory::create(&service_name);
+                        match sut {
+                            Ok(_) => {
+                                success_counter.fetch_add(1, Ordering::Relaxed);
+                            }
+                            Err(e) => {
+                                error_check(e);
+                            }
+                        }
+
+                        barrier_exit.wait();
+                    }
+                }));
+            }
+
+            for thread in threads {
+                thread.join().unwrap();
+            }
+
+            assert_that!(success_counter.load(Ordering::Relaxed), eq NUMBER_OF_ITERATIONS as u64)
+        });
+    }
+
+    #[test]
+    fn concurrent_creating_services_with_same_name_fails_for_all_but_one<Sut: Service>() {
+        concurrent_creating_services_with_same_name_fails_for_all_but_one_impl::<
+            publish_subscribe::PortFactory<Sut, u64>,
+            _,
+        >(|error| {
+            assert_that!(error, any_of [PublishSubscribeCreateError::AlreadyExists,
+                         PublishSubscribeCreateError::IsBeingCreatedByAnotherInstance,
+            ]);
+        });
+
+        concurrent_creating_services_with_same_name_fails_for_all_but_one_impl::<
+            event::PortFactory<Sut>,
+            _,
+        >(|error| {
+            assert_that!(error, any_of [EventCreateError::AlreadyExists,
+                         EventCreateError::IsBeingCreatedByAnotherInstance,
+            ]);
+        });
+    }
+
+    fn concurrent_opening_and_closing_services_with_same_name_is_handled_gracefully_impl<
+        Factory: SutFactory,
+        F: Sync,
+    >(
+        error_check: F,
+    ) where
+        F: Fn(Factory::OpenError),
+    {
+        let _watch_dog = Watchdog::new_with_timeout(Duration::from_secs(30));
+        const NUMBER_OF_CLOSE_THREADS: usize = 1;
+        const NUMBER_OF_OPEN_THREADS: usize = 4;
+        const NUMBER_OF_THREADS: usize = NUMBER_OF_CLOSE_THREADS + NUMBER_OF_OPEN_THREADS;
+        const NUMBER_OF_ITERATIONS: usize = 50;
+
+        let barrier_enter = Barrier::new(NUMBER_OF_THREADS);
+        let barrier_exit = Barrier::new(NUMBER_OF_THREADS);
+        let service_name = generate_name();
+
+        std::thread::scope(|s| {
+            let mut threads = vec![];
+            threads.push(s.spawn(|| {
+                for _ in 0..NUMBER_OF_ITERATIONS {
+                    let sut = Factory::create(&service_name).unwrap();
+                    barrier_enter.wait();
+
+                    drop(sut);
+
+                    barrier_exit.wait();
+                }
```

Comment:
@elfenpiff it seems the creation of the service fails with sporadically with `AlreadyExists` when there are more threads than cores. Is there already a way to clean everything up?

Reply:
@elBoberido No, sadly, not yet. It would belong into `iceoryx2/src/service/mod.rs` but it comes with some challenges. So first, the static storage must become inaccessible, then the dynamic storage of the service has to be removed, and finally, the static storage.

And this has to be tested concurrently so that when one tries to open a service that is currently removed, either the remove fails with still in-use failure or the open fails with no longer exists. Maybe it makes then also sense to provide a `force_remove` method that does not care for anything and just removes the artifacts in the order above.

But when you think further, the pair messaging-pattern/service-name is unique in the system, so with remove you have to provide the messaging pattern as well.
Maybe it makes more sense to add some kind of builder and remove a service like:
```rust
Service::event().remove("some_name");
Service::publish_subscribe().force_remove("some_name");
```

Reply:
Okay, then I'll skip the test for now